components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,684 @@
+    }
+
+    @Override
+    public String getName() {
+
+        return jwtInboundAuthConfig.getName();
+    }
+
+    @Override
+    public String getCallbackPath(IdentityMessageContext identityMessageContext) {
+
+        String endpoint = "identity";
+        String tenantDomain = identityMessageContext.getRequest().getTenantDomain();
+        if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+            endpoint = "t/" + tenantDomain + "/identity";
+        }
+        return IdentityUtil.getServerURL(endpoint, false, false);
+    }
+
+    @Override
+    public String getRelyingPartyId() {
+
+        return this.relyingParty;
+    }
+
+    @Override
+    public String getRelyingPartyId(IdentityMessageContext identityMessageContext) {
+
+        return this.relyingParty;
+    }
+
+    @Override
+    public boolean canHandle(IdentityRequest identityRequest) {
+
+        if (identityRequest instanceof JWTInboundRequest) {
+            this.relyingParty = identityRequest.getParameter(JWTInboundConstants.SP_ID);
+            if (log.isDebugEnabled()) {
+                log.debug("Relying party: " + JWTInboundUtil.neutralize(this.relyingParty));
+            }
+        }
+        return StringUtils.isNotBlank(this.relyingParty);
+    }
+
+    /**
+     * This method represents the bulk of the functionality, where the developer chooses what should take place when
+     * the authentication request reaches the processor. In this instance, a check is done first to determine if the
+     * request is coming new from the /identity servlet.
+     *
+     * @param identityRequest the request object (or a subclass of it), which can be coming either from the /identity
+     *                        servlet or from the framework after authentication).
+     * @return an instance of IdentityResponse which may be further customised (similar to how an IdentityRequest can
+     * be customised).
+     * @throws FrameworkException if any abnormal conditions are encountered by the framework during authentication.
+     */
+    public IdentityResponse.IdentityResponseBuilder process(IdentityRequest identityRequest) throws FrameworkException {
+
+        IdentityMessageContext messageContext = new IdentityMessageContext<>(identityRequest,
+                new HashMap<String, String>());
+        JWTInboundResponseBuilder respBuilder = new JWTInboundResponseBuilder(messageContext);
+
+        String sessionId = identityRequest.getParameter(InboundConstants.RequestProcessor.CONTEXT_KEY);
+        String logoutRequestPath = JWTInboundConstants.BASE_PATH + JWTInboundConstants.LOGOUT_PATH;
+
+        if (isRelyingPartyExist(identityRequest)) {
+
+            // Validate endpoint URL.
+            if (!validateEndpointUrl(identityRequest)) {
+                String msg = "Mandatory configuration: Endpoint API is not configured.";
+                log.error(msg, new JWTIdentityException(msg));
```

Comment:
Add a SP identifier to all the errors where it is available.

Reply:
Fixed